src/ipc/ipc4/handler.c

```diff
@@ -290,6 +290,9 @@ static int set_pipeline_state(uint32_t id, uint32_t cmd, bool *delayed)
 			}
 			if (ret == PPL_STATUS_SCHEDULED)
 				*delayed = true;
+			break;
+		default:
+			ret = 0;
```

Comment:
sure that default is "0 == SUCCESS"? in case a state has not been found maybe better to return some error code?

Reply:
I wanted to keep identical behavior. Don't know whether not hitting any of these statuses is wrong since we have more states defined than these under switch.

Reply:
I think we should have a IPC4_SUCESS value here ? and maybe it's 0. Btw, do we need a break or fallthough here ?
```diff
diff --git a/src/ipc/ipc4/handler.c b/src/ipc/ipc4/handler.c
index 35f9361ac..2a4baf4b5 100644
--- a/src/ipc/ipc4/handler.c
+++ b/src/ipc/ipc4/handler.c
@@ -309,6 +309,7 @@ static int set_pipeline_state(struct ipc_comp_dev *ppl_icd, uint32_t cmd,
        case SOF_IPC4_PIPELINE_STATE_EOS:
                if (status != COMP_STATE_ACTIVE)
                        return IPC4_INVALID_REQUEST;
+               /* fallthrough */
        case SOF_IPC4_PIPELINE_STATE_SAVED:
        case SOF_IPC4_PIPELINE_STATE_ERROR_STOP:
        default:
```

Reply:
break or fallthrough, 0 or IPC4_SUCCESS - just a code style. 

My concern is if it should not be like:
``` 
                case OTHER_ALLOWED_STATE1:
                case OTHER_ALLOWED_STATE2:
                       ret = 0;
                       break;
		default:
			tr_err(&ipc_tr, "ipc: current status %d", status);
			return IPC4_INVALID_REQUEST;
		}
```

Reply:
ok, @jxstelter can we look at the fallthrough/break update in a new PR.